plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/ProxySettings.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.azure;
+
+import com.azure.core.http.ProxyOptions;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.SettingsException;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+public class ProxySettings {
+
+    public static final ProxySettings NO_PROXY_SETTINGS = new ProxySettings(ProxyType.NONE, null, -1, null, null);
+
+    private final ProxyType type;
+
+    private final InetAddress host;
+
+    private final String username;
+
+    private final String password;
+
+    private final int port;
+
+    public static enum ProxyType {
+        HTTP(ProxyOptions.Type.HTTP.name()),
+
+        SOCKS4(ProxyOptions.Type.SOCKS4.name()),
```

Comment:
We probably need `SOCKS(ProxyOptions.Type.SOCKS4.name())` since it was a legit proxy value before